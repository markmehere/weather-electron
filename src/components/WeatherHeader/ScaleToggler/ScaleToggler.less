.toggler div {
  display: inline-block;
  color: #aaa;
}

.centerToggle {
  margin: 12px 24px;
  cursor: pointer;
}
