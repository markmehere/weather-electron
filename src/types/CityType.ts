export type CityType = {
  value: number; /* woeid */
  label: string; /* city, country code (e.g. Melbourne, AU) */
};