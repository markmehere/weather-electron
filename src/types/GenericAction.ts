import { Action } from "redux";

export interface GenericAction<T> extends Action {
  payload: T;
} 
