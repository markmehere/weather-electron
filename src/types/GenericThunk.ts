import { ThunkAction } from "redux-thunk";
import { AppStore } from "../redux/reducers";

export type GenericThunk = ThunkAction<any, AppStore, undefined, any>;