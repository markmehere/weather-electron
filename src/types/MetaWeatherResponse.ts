import { WeatherKind } from "./WeatherKind";

export interface ConsolidatedWeatherResponse {
  id: string;
  weather_state_name: string;
  weather_state_abbr: WeatherKind;
  created: string;
  applicable_date: string;
  min_temp: number;
  max_temp: number;
  wind_speed: number;
  wind_direction: number;
  air_pressure: number;
  humidity: number;
  visibility: number;
  predictability: number;
}

export interface MetaWeatherResponse {
  consolidated_weather: ConsolidatedWeatherResponse[]
}