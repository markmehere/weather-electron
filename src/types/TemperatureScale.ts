export enum TemperatureScale {
  Celsius,
  Fahrenheit
}
