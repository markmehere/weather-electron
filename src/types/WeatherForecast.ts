import { WeatherKind } from "./WeatherKind";

export interface WeatherForecast {
  max: number;
  min: number;
  date: string;
  kind: WeatherKind;
}
