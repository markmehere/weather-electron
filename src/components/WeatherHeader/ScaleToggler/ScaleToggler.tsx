import React from "react";
import ToggleButton from "react-toggle-button";
import styles from "./ScaleToggler.less";
import { TemperatureScale } from "../../../types/TemperatureScale";

export interface ScaleTogglerProps {

  // The current temperature scale
  value: TemperatureScale;

  // Callback when temperature is changed
  onToggle: (value: TemperatureScale) => void;

}

const trackStyle = {
  height: 15,
};

const thumbStyle = {
  position: 'absolute',
  width: 30,
  height: 30,
  boxShadow: `0 0 2px rgba(0,0,0,.12),0 2px 4px rgba(0,0,0,.24)`
};

export class ScaleToggler extends React.Component<ScaleTogglerProps> {

  render() {
    const { value, onToggle } = this.props;

    return  <div className={styles.toggler}>
      <div>&deg;C</div>
      <div className={styles.centerToggle}>
        <ToggleButton
          inactiveLabel={''}
          activeLabel={''}
          value={ value === TemperatureScale.Fahrenheit }
          onToggle={(value: boolean) => {
            onToggle(value ? TemperatureScale.Celsius : TemperatureScale.Fahrenheit );
          }}
          colors={{
            activeThumb: {
              base: 'rgb(250,250,250)',
            },
            inactiveThumb: {
              base: 'rgb(62,130,247)',
            },
            active: {
              base: 'rgb(207,221,245)',
              hover: 'rgb(177, 191, 215)',
            },
            inactive: {
              base: 'rgb(65,66,68)',
              hover: 'rgb(95,96,98)',
            }
          }}
          trackStyle={trackStyle}
          thumbStyle={thumbStyle}
          thumbAnimateRange={[-10, 36]}
          />
      </div>
      <div>&deg;F</div>
    </div>;
  }
}