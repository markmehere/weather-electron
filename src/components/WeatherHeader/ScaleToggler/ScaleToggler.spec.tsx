import React from 'react';
import { shallow } from 'enzyme';
import { ScaleToggler } from './ScaleToggler';
import { TemperatureScale } from '../../../types/TemperatureScale';

describe('<ScaleToggler />', () => {

  const defaultProps = {
    value: TemperatureScale.Celsius,
    onToggle: jest.fn()
  }

  it('should render with city selector', () => {
    const result = shallow(<ScaleToggler {...defaultProps}  />);
    expect(result.text()).toContain('\u00B0C');
    expect(result.text()).toContain('\u00B0F');
  });

  it('matches snapshot', () => {
    const result = shallow(<ScaleToggler {...defaultProps}  />);
    expect(result).toMatchSnapshot();
  });

});