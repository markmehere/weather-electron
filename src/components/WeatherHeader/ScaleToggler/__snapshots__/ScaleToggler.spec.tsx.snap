// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScaleToggler /> matches snapshot 1`] = `
<div>
  <div>
    °C
  </div>
  <div>
    <Toggle
      activeLabel=""
      activeLabelStyle={Object {}}
      activeLabelStyleHover={Object {}}
      animateThumbStyleHover={[Function]}
      animateThumbStyleToggle={[Function]}
      animateTrackStyleHover={[Function]}
      animateTrackStyleToggle={[Function]}
      colors={
        Object {
          "active": Object {
            "base": "rgb(207,221,245)",
            "hover": "rgb(177, 191, 215)",
          },
          "activeThumb": Object {
            "base": "rgb(250,250,250)",
          },
          "inactive": Object {
            "base": "rgb(65,66,68)",
            "hover": "rgb(95,96,98)",
          },
          "inactiveThumb": Object {
            "base": "rgb(62,130,247)",
          },
        }
      }
      containerStyle={Object {}}
      inactiveLabel=""
      inactiveLabelStyle={Object {}}
      inactiveLabelStyleHover={Object {}}
      internalHoverSpringSetting={
        Object {
          "damping": 20,
          "stiffness": 180,
        }
      }
      internalSpringSetting={
        Object {
          "damping": 22,
          "stiffness": 180,
        }
      }
      onToggle={[Function]}
      passThroughInputProps={Object {}}
      thumbAnimateRange={
        Array [
          -10,
          36,
        ]
      }
      thumbStyle={
        Object {
          "boxShadow": "0 0 2px rgba(0,0,0,.12),0 2px 4px rgba(0,0,0,.24)",
          "height": 30,
          "position": "absolute",
          "width": 30,
        }
      }
      thumbStyleHover={Object {}}
      trackStyle={
        Object {
          "height": 15,
        }
      }
      trackStyleHover={Object {}}
      value={false}
    />
  </div>
  <div>
    °F
  </div>
</div>
`;
